src/librustc_resolve/diagnostics.rs

```diff
@@ -93,6 +93,41 @@ mod bar {
 ```
 "##,
 
+E0254: r##"
+This error indicates that an imported symbol resolved to an already imported
+symbol.
+
+An example of this error:
+
+```
+// crate foo
+pub use foo::bar;
+pub mod foo; // note that there's a `pub` here
+```
+
+```
+// other program
+extern crate foo;
+
+use foo::*; // error, `foo::*` tries to bind `foo::foo` to `foo`
```

Comment:
(i'm investigating this now; I can see that the most trivial examples of duplicate imported symbols flag *other* error codes, so let me see if I can tease out exactly what it is about your example that matters)